pyfar/dsp/dsp.py

```diff
@@ -1487,4 +1494,314 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1):
+        not affect single-channel signals.
+
+        ``'each'``
+            Separate normalization of each channel.
+        ``'max'``
+            Normalize to the maximum `reference` value across channels.
+        ``'min'``
+            Normalize to the minimum `reference` value across channels.
+        ``'mean'``
+            Normalize to the mean `reference` value across the channels.
+
+       The default is ``'max'``
+    target: scalar, array
+        The target to which the signal is normalized in the unit according to
+        `dB`. Can be a scalar or an array. In the latter case the shape of
+        `target` must be broadcasted ``signal.cshape``. The default is 0 dB if
+        ``dB=True`` and 1 otherwise.
+    freq_range: tuple, array_like, double
+        Two element vector specifying upper and lower frequency in Hz between
+        which the `reference` value for normalization is computed. This will
+        just effect if the normalization is based on the frequency data.
+        Therefore, `freq_range` won't be used if ``domain='time'``.
+        The default is ``None``, which will use the whole frequency range of
+        the signal.
+
+    Returns
+    --------
+    normalized_signal: Signal, TimeData, FrequencyData
+        The normalized signal
+    return_reference: bool
+        If ``return_reference=True`` the function also return the `reference`
+        values described above.
+    """
+
+    # check input
+    if not isinstance(signal, (pyfar.Signal, pyfar.FrequencyData,
+                               pyfar.TimeData)):
+        raise TypeError(("Input data has to be of type 'Signal', 'TimeData' "
+                         "or 'FrequencyData'."))
+    if domain not in ('time', 'freq'):
+        raise ValueError("domain must be 'time' or 'freq'.")
+    if type(signal) == pyfar.classes.audio.FrequencyData and domain == 'time':
+        raise ValueError((
+            f"domain is '{domain}' and signal is type '{signal.__class__}'"
+            " but must be of type 'Signal' or 'TimeData'."))
+    if type(signal) == pyfar.classes.audio.TimeData and domain == 'freq':
+        raise ValueError((
+            f"domain is '{domain}' and signal is type '{signal.__class__}'"
+            " but must be of type 'Signal' or 'FrequencyData'."))
+    if len(limits) != 2:
+        raise ValueError("limits must be an array like of length 2.")
+    if tuple(limits) != (None, None) and \
+            operation in ('energy', 'power', 'rms'):
+        raise ValueError(
+            f"limits must be (None, None) if operation is {operation}")
+    if (domain == "time" and unit not in ("s", "samples", None)) or \
+            (domain == "freq" and unit not in ("hz", "bins", None)):
+        raise ValueError(f"'{unit}' is an invalid unit for domain {domain}")
+
+    # get and check the limits
+    find = signal.find_nearest_time if domain == "time" \
+        else signal.find_nearest_frequency
+
+    if unit in ("hz", "s"):
```

Comment:
What about the different time units? See e.g. ``plot.group_delay``
```suggestion
    if unit in ("Hz", "s"):
```